### PR TITLE
[android] Fixed localized Feature type.

### PR DESCRIPTION
**Pull request by vng**
Closes https://github.com/organicmaps/organicmaps/issues/7211

@Jean-BaptisteC @dvdmrtnz Please, take a look at todo.

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/search/SearchResult.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public class SearchResult
   public static class Description
   {
     public final FeatureId featureId;
-    public final String featureType;
+    public final String featureType;  // localized
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/search/SearchResult.java:44]:
```suggestion
    public final String localizedFeatureType;
```

File: android/app/src/main/java/app/organicmaps/search/SearchFragment.java [hunk 2]
```diff
@@ -398,9 +398,8 @@ void showSingleResultOnMap(@NonNull SearchResult result, int resultIndex)
     if (RoutingController.get().isWaitingPoiPick())
```
Comment (dvdmrtnz) [inline android/app/src/main/java/app/organicmaps/search/SearchFragment.java:398]: What is a `WAITING_POI_PICK` and when is `RoutingController.get().isWaitingPoiPick()` true?

Comment (vng) [reply android/app/src/main/java/app/organicmaps/search/SearchFragment.java:398]: When building a route and finish point is defined by search?

File: android/app/src/main/java/app/organicmaps/search/SearchFragment.java [hunk 3]
```diff
@@ -398,9 +398,8 @@ void showSingleResultOnMap(@NonNull SearchResult result, int resultIndex)
...
     {
       SearchResult.Description description = result.description;
-      String subtitle = description != null
-          ? Utils.getLocalizedFeatureType(requireContext(), description.featureType)
-          : "";
+      String subtitle = (description != null) ? description.featureType : "";
+      /// @todo Hm, is it ok here? Empty title if name is not empty?
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/search/SearchFragment.java:402]: How to reproduce it?

Comment (vng) [reply android/app/src/main/java/app/organicmaps/search/SearchFragment.java:402]: Building a route and start/finish point are defined with search.

File: android/app/src/main/java/app/organicmaps/search/SearchFragment.java [hunk 4]
```diff
@@ -398,9 +398,8 @@ void showSingleResultOnMap(@NonNull SearchResult result, int resultIndex)
...
       String title = TextUtils.isEmpty(result.name) ? subtitle : "";
```
Comment (dvdmrtnz) [inline android/app/src/main/java/app/organicmaps/search/SearchFragment.java:403]:
```suggestion
      String title = TextUtils.isEmpty(result.name) ? subtitle : result.name;
```

Should use the name when name is not empty

**Review by dvdmrtnz — APPROVED** [review]

**Pull request merged**